src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/AuditWhitelistManager.java

```diff
@@ -61,89 +55,96 @@
 
-            addStatements.forEach(
-                    (o, r) -> whitelistDataAccess.addToWhitelist(role, o, r));
+        if (whitelistOperation == WhitelistOperation.DROP_LEGACY)
+        {
+            dropRoleOption(performer, role, optionEntry);
+        }
+        else
+        {
+            alterWhitelistOption(performer, role, whitelistOperation, optionEntry);
         }
     }
 
-    void alterRoleWhitelist(AuthenticatedUser performer, RoleResource role, RoleOptions options)
+    private void dropRoleOption(AuthenticatedUser performer, RoleResource role, Map.Entry<String, String> optionEntry)
     {
-        if (options.getCustomOptions().isPresent())
-        {
-            Map<String, Set<IResource>> addStatements = new HashMap<>();
-            Map<String, Set<IResource>> removeStatements = new HashMap<>();
-            for (Map.Entry<String, String> optionEntry : options.getCustomOptions().get().entrySet())
-            {
-                WhitelistOperation whitelistOperation = whitelistOptionParser.parseWhitelistOperation(optionEntry.getKey());
-                Set<IResource> resources = whitelistOptionParser.parseResource(optionEntry.getValue());
-                String operation = whitelistOptionParser.parseTargetOperation(optionEntry.getKey());
-
-                checkPermissionToWhitelist(performer, resources);
-
-                if (whitelistOperation == WhitelistOperation.GRANT)
-                {
-                    addStatements.put(operation, resources);
-                }
-                else
-                {
-                    removeStatements.put(operation, resources);
-                }
-            }
-
-            addStatements.forEach(
-                    (o, r) -> whitelistDataAccess.addToWhitelist(role, o, r));
-            removeStatements.forEach(
-                    (o, r) -> whitelistDataAccess.removeFromWhitelist(role, o, r));
-        }
+        whitelistOptionParser.parseDropValue(optionEntry.getValue());
+        checkPermissionToDropLegacyTable(performer, role);
+        whitelistDataAccess.dropLegacyWhitelistTable();
     }
 
-    Map<String, Set<IResource>> getRoleWhitelist(RoleResource role)
+    private void alterWhitelistOption(AuthenticatedUser performer, RoleResource role, WhitelistOperation whitelistOperation, Map.Entry<String, String> optionEntry)
     {
-        Map<String, Set<IResource>> daoWhitelist = whitelistDataAccess.getWhitelist(role);
-        Set<IResource> resources = daoWhitelist.get(OPERATION_ALL);
-        if (resources != null)
+        IResource resource = whitelistOptionParser.parseResource(optionEntry.getValue());
+        Set<Permission> operations = whitelistOptionParser.parseTargetOperation(optionEntry.getKey(), resource);
+
+        whitelistContract.verify(operations, resource);
+        checkPermissionToWhitelist(performer, resource);
+
+        if (whitelistOperation == WhitelistOperation.GRANT)
```

Comment:
Perhaps

if (grant)
..
else if(revoke)
...
else
throw unexcpctedoperation

in case new operations are added.